src/main/java/dev/joopie/jambot/music/command/PlayCommandHandler.java

```diff
@@ -70,69 +71,91 @@ public boolean shouldHandle(final CommandInteractionPayload event) {
-            parsedVideoId = YOUTUBE_URL + videoId;
-        }
+        var parsedVideoId = YOUTUBE_URL + videoId;
+            // Create and send the message with buttons
+            return event.getHook().sendMessage(
+                            "**Track added!**%n%n OK, we added the track to the queue! In order to improve our service we would like%n to ask you to rate our search result.%n%n Please let us know :thumbsup_tone3: or :thumbsdown_tone3: if we got the correct result for you.%n%n %s".formatted(parsedVideoId))
+                    .addActionRow(
+                            Button.success("accept", "Accept").withEmoji(Emoji.fromUnicode("U+1F44D U+1F3FD")),
+                            Button.danger("reject", "Reject").withEmoji(Emoji.fromUnicode("U+1F44E U+1F3FD"))
+                    );
 
-        // Create and send the message with buttons
-        return event.getHook().sendMessage(
-                    "**Track added!**%n%n OK, we added the track to the queue! In order to improve our service we would like%n to ask you to rate our search result.%n%n Please let us know :thumbsup_tone3: or :thumbsdown_tone3: if we got the correct result for you.%n%n %s".formatted(parsedVideoId))
-                .addActionRow(
-                        Button.success("accept", "Accept").withEmoji(Emoji.fromUnicode("U+1F44D U+1F3FD")),
-                        Button.danger("reject", "Reject").withEmoji(Emoji.fromUnicode("U+1F44E U+1F3FD"))
-                );
-        
     }
 
+    public WebhookMessageCreateAction<Message> handlePlay(CommandInteraction event, String videoId) {
+        var mediaUrl = videoId;
+        if (videoId.length() == 11) {
+            mediaUrl = YOUTUBE_URL + videoId;
+        }
+        musicService.play(event.getMember(), videoId);
+        return event.getHook().sendMessage(
+                        "**Track added!**%n%n OK, we added the track to the queue! %s".formatted(mediaUrl));
+
+    }
 
     private String handleSpotifyLink(String input) {
         if (input.contains("track")) {
-            final Optional<Track> spotifyResult = searchService.getTrack(input);
+            final var spotifyResult = searchService.getTrack(input);
             return spotifyResult.map(searchService::performYoutubeSearch).orElse(null);
         }
 
         return Strings.EMPTY;
     }
+
+    private String extractYouTubeVideoId(String url) {
+        var youTubeLinkWithoutProtocolAndDomain = removeProtocolAndDomain(url);
+
+        for (var pattern : VIDEO_ID_PATTERNS) {
+            var matcher = pattern.matcher(youTubeLinkWithoutProtocolAndDomain);
+
+            if (matcher.find()) {
+                return matcher.group(1);
+            }
+        }
+
+        return null;
+    }
+
+    private String removeProtocolAndDomain(String url) {
+        var matcher = YOU_TUBE_URL_PATTERN.matcher(url);
+
+        if (matcher.find()) {
+            return url.replace(matcher.group(), "");
+        }
+        return url;
+    }
```

Comment:
Copy/Paste implementation which just works. Don't want to spend more on regex. Suggestions? Put them here and I will implement them

Reply:
Primaaaaa